jQuery.merge: add note about it correcting the `length`

arthurvr opened the pull request:
Fixes https://github.com/jquery/api.jquery.com/issues/686

arthurvr commented:
@mzgol WDYT?

mzgol commented:
The page still mentions that `jQuery.merge` merges two arrays (and does it in multiple places). This should all be changed to mention array-likes. Apart from that, the change looks fine (modulo the outcome of https://github.com/jquery/api.jquery.com/issues/686#issuecomment-90633413).

arthurvr commented:
So, I updated the entry to use the `array-like object` type so we don't need to get funky explaining why we correct the length. What do we think?

When updating it I noticed a little issue with https://github.com/jquery/api.jquery.com/commit/929f9e2472ec0da31446860f3e6a34f35321a6a2, I called the type `array-like objects` but it should be `array-like object`. Sorry! Fixed that in a separated commit.

mzgol commented:
I don't see it mentioned anywhere in the docs that if jQuery modifies an array-like object, its length gets corrected. This would fit perfectly here, though so I'd leave this text.

arthurvr commented:
@mzgol Updated that. Thanks! At first it sounded quite logical to me so I didn't find an explicit note necessary, but yeah, you're right. WDYT?

gibson042 commented on entries/jQuery.merge.xml:
```diff
@@ -19,6 +19,7 @@ var newArray = $.merge([], oldArray);
     </code></pre>
     <p>This shortcut creates a new, empty array and merges the contents of oldArray into it, effectively cloning the array.</p>
     <p>Prior to jQuery 1.4, the arguments should be true Javascript Array objects; use <code>$.makeArray</code> if they are not.</p>
+    <p>Because <code>jQuery.merge()</code> is capable of merging array-like objects it will correct the <code>length</code> property.</p>
```
Missing a comma before "it". But this note might be better placed near "is destructive" anyway, something like "It alters the <code>length</code> and numeric index properties of the first parameter to include items from the second."

arthurvr replied:
@gibson042 Sounds good. PTAL.

gibson042 commented:
LGTM.
